### PR TITLE
Remove [<unit>] from C usage messages

### PR DESCRIPTION
**Pull request by PaulWessel**
**Description of proposed changes**

To match the changes in RST, here we remove optional unit indicators in the C code.

### CODE REVIEW

File: src/gmt_init.c [hunk 1]
```diff
@@ -1322,7 +1322,7 @@ GMT_LOCAL int gmtinit_parse_f_option (struct GMT_CTRL *GMT, char *arg) {
 		pos = 1;
 		start = stop = 1;
 	}
-	else if (copy[0] == 'g' || copy[0] == 'p') {	/* Got -f[i|o]g which is shorthand for -f[i|o]0x,1y, or -fp[<unit>] (see below) */
+	else if (copy[0] == 'g' || copy[0] == 'p') {	/* Got -f[i|o]g which is shorthand for -f[i|o]0x,1y, or -fp (see below) */
```
Comment (seisman) [inline src/gmt_init.c]: Keep this <unit>?

Comment (PaulWessel) [reply src/gmt_init.c]: Yes, done

File: src/mapproject.c [hunk 2]
```diff
@@ -189,7 +189,7 @@ GMT_LOCAL int usage (struct GMTAPI_CTRL *API, int level) {
 	const char *name = gmt_show_name_and_purpose (API, THIS_MODULE_LIB, THIS_MODULE_CLASSIC_NAME, THIS_MODULE_PURPOSE);
 	if (level == GMT_MODULE_PURPOSE) return (GMT_NOERROR);
 	GMT_Message (API, GMT_TIME_NONE, "usage: %s <table> %s %s\n", name, GMT_J_OPT, GMT_Rgeo_OPT);
-	GMT_Message (API, GMT_TIME_NONE, "\t[-Ab|B|f|F|o|O[<lon0>/<lat0>][+v]] [-C[<dx></dy>]] [-D%s] [-E[<datum>]] [-F[<unit>]]\n\t[-G[<lon0>/<lat0>][+a][+i][+u<unit>][+v]]", GMT_DIM_UNITS_DISPLAY);
+	GMT_Message (API, GMT_TIME_NONE, "\t[-Ab|B|f|F|o|O[<lon0>/<lat0>][+v]] [-C[<dx></dy>]] [-D%s] [-E[<datum>]] [-F]\n\t[-G[<lon0>/<lat0>][+a][+i][+u<unit>][+v]]", GMT_DIM_UNITS_DISPLAY);
```
Comment (seisman) [inline src/mapproject.c]: Add this one back.

File: src/gmt_init.c [hunk 3]
```diff
@@ -1334,7 +1334,7 @@ GMT_LOCAL int gmtinit_parse_f_option (struct GMT_CTRL *GMT, char *arg) {
 		pos = 1;
 		start = stop = 1;
 	}
-	if (copy[0] == 'p') {	/* Got -f[i|o]p[<unit>] for projected floating point map coordinates (e.g., UTM meters) */
+	if (copy[0] == 'p') {	/* Got -f[i|o]p for projected floating point map coordinates (e.g., UTM meters) */
```
[the remark this answers is not in the document]
Comment (PaulWessel) [reply src/gmt_init.c]: Done

**Pull request merged**